week7/src/pages/EditPage.tsx

```diff
@@ -0,0 +1,30 @@
+import LetterForm from '../components/LetterForm';
+import styled from 'styled-components';
+import { useLocation } from 'react-router-dom';
+import { Letter } from '../types/Letter';
+
+export default function EditPage() {
+  const location = useLocation();
+  const letterInfo = location.state as Letter;
```

Comment:
```suggestion
  const letterInfo = location.state;

  const isLetterState = (state: unknown): state is Letter => {
    if (typeof state !== 'object' || state === null) return false;
    
    const letterField = ['_id', 'password', 'hint', 'content', 'images', '__v', 'name'];
    const keys = Object.keys(state);
    
    return keys.every((key) => letterField.includes(key));
  }

  if (!isLetterState(letterInfo)) return;
```

이런 식으로 커스텀 타입가드로 내로잉 할 수도 있을 듯! 
항상 맞는 방법은 아닐수도 있지만..

Reply:
이런 방법도 있군아
참고해서 해볼게 고마워..................